### PR TITLE
fix: cross-platform `--workspace` support

### PR DESCRIPTION
**Pull request by rvolosatovs**
Fix-up Windows and Mac support broken by #2419

### CODE REVIEW

File: tests/crates/enarx_exec_tests/src/lib.rs [hunk 1]
```diff
@@ -8,6 +8,7 @@ macro_rules! musl_fsbase_fix {
         /// Set FSBASE
         ///
         /// Overwrite the only location in musl, which uses the `arch_prctl` syscall
+        #[cfg(target_os = "linux")]
```
Comment (haraldh) [inline tests/crates/enarx_exec_tests/src/lib.rs:11]:
```suggestion
        #[cfg(all(test, target_arch = "x86_64",  target_vendor = "unknown", target_os = "linux", target_env = "musl"))]
```

File: tests/crates/enarx_syscall_tests/src/lib.rs [hunk 2]
```diff
@@ -109,11 +109,12 @@ pub struct Args {
     pub arg5: usize,
 }
 
-pub fn syscall(nr: i64, args: Args) -> (usize, usize) {
+#[cfg(target_vendor = "unknown")]
```
Comment (haraldh) [inline tests/crates/enarx_syscall_tests/src/lib.rs:112]:
```suggestion
#[cfg(all(target_arch = "x86_64", target_vendor = "unknown"))]
```

File: tests/crates/enarx_syscall_tests/src/lib.rs [hunk 3]
```diff
@@ -129,6 +130,11 @@ pub fn syscall(nr: i64, args: Args) -> (usize, usize) {
     (rax, rdx)
 }
 
+#[cfg(not(target_vendor = "unknown"))]
```
Comment (haraldh) [inline tests/crates/enarx_syscall_tests/src/lib.rs:133]:
```suggestion
#[cfg(not(all(target_arch = "x86_64", target_vendor = "unknown")))]
```

**Review by haraldh — CHANGES_REQUESTED** [review]

File: tests/crates/enarx_exec_tests/src/lib.rs (hunk 1 above)
Comment (rvolosatovs) [reply tests/crates/enarx_exec_tests/src/lib.rs:11]: I don't think `test` is a.) necessary, b.) actually used, since this is a bindep

**Review by haraldh — APPROVED** [review]

**Pull request merged**